src/ExtractTileOperations.cpp

```diff
@@ -197,14 +369,24 @@ Matmul convert_to_matmul(const Store *op, const string &new_name, AMXOpType op_t
 
     const auto *lhs_load = matches[0].as<Load>();
     const auto *rhs_broadcast = matches[1].as<Broadcast>();
-    if (!lhs_load || !rhs_broadcast) {
+
+    const Cast *rhs_cast = nullptr;
+
+    if (lhs_load && !rhs_broadcast) {
+        // now working on a larger k dimension
```

Comment:
I don't understand this comment. It seems to describe the PR rather than the code here.

Reply:
I've added a slightly larger description here 7be1b5b6867ce22ae36bd4f15e0f07545f3c5c81.
And with further explanation for the layout of the matrices a80fc766ff63a2002a4fc459c04b17ef0127d784.